hat/assets/js/apps/Iaso/domains/forms/detail.tsx

```diff
@@ -205,6 +217,21 @@ const FormDetail: FunctionComponent<FormDetailProps> = ({ router, params }) => {
             setFormState(formatFormData(form));
         }
     }, [form, setFormState]);
+
+    useEffect(() => {
+        if (
+            currentForm.period_type.value === NO_PERIOD ||
+            !currentForm.period_type.value
+        ) {
+            setDetailRequiredFields(
+                requiredFields.filter(
+                    field => field.key !== 'single_per_period',
+                ),
+            );
+        } else {
+            setDetailRequiredFields(requiredFields);
+        }
+    }, [currentForm.period_type.value]);
```

Comment:
target is to get detailRequiredFields if i get it well, why don't you use a `usemo` for this, instead of setting the state each time and force a re-render ?

Reply:
Good remark. Thank you